lib/datadog/tracing/span.rb

```diff
@@ -204,6 +204,13 @@ def pretty_print(q)
         end
       end
 
+      # Single Span Sampling has chosen to keep this span
+      # regardless of the trace-level sampling decision
+      # @!visibility private
```

Comment:
If this is meant to be private, can we move it behind the `private` keyword? I don't like soft rules like that which can be easily violated if you didn't read the docs.

Reply:
My main concern is performance:
```
Warming up --------------------------------------
             obj.foo     1.421M i/100ms
      obj.send(:foo)   363.874k i/100ms
Calculating -------------------------------------
             obj.foo     17.501M (± 2.3%) i/s -     53.987M in   3.086433s
      obj.send(:foo)     10.233M (± 3.0%) i/s -     30.929M in   3.025515s

Comparison:
             obj.foo: 17500810.6 i/s
      obj.send(:foo): 10232543.6 i/s - 1.71x  (± 0.00) slower
```

And how much of a sore `obj.send(:foo)` looks compared to `obj.foo`.

I'll change this method to be really private and use `obj.send(:foo)` when calling it in this PR.